lib/google/youtube/utils.py

```diff
@@ -0,0 +1,77 @@
+# Python Standard Library Imports
+import re
+import urllib
+
+# Third Party (PyPI) Imports
+import requests
+import rollbar
+
+# Django Imports
+from django.conf import settings
+
+
+def extract_youtube_video_id(youtube_video_url):
+    try:
+        # Ref to build regex: https://gist.github.com/rodrigoborgesdeoliveira/987683cfbfcc8d800192da1e73adc486
+        re_match = re.search(
+            r'(?:v=|youtu\.be\/|\/v\/|\/e\/|\/watch\?v=|\/watch\?si=|\/watch\?)([0-9A-Za-z_-]+)',
+            youtube_video_url,
+        )
+    except Exception as e:
+        youtube_video_id = None
+        rollbar.report_message(
+            'Failed to extract video ID from YouTube URL',
+            extra_data={
+                'video_url': youtube_video_url,
+                'error': f'{e}',
+            },
+        )
+    else:
+        youtube_video_id = re_match.group(1) if re_match else None
+
+    return youtube_video_id
+
+
+def build_youtube_api_url(youtube_video_id):
+    base_url = 'https://www.googleapis.com/youtube/v3/videos?'
```

Comment:
Define in `lib/google/youtube/constants.py` as 

```
GOOGLE_APIS_YOUTUBE_BASE_URL = 'https://www.googleapis.com/youtube/v3'
GOOGLE_APIS_YOUTUBE_VIDEOS_URL = f'{GOOGLE_APIS_YOUTUBE_BASE_URL}/videos'
```

Reply:
Done